zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template/proposal.rs

```diff
@@ -42,23 +42,22 @@ pub enum ProposalResponse {
 
 impl ProposalResponse {
     /// Return a rejected response containing an error kind and detailed error info.
-    pub fn rejected<S: ToString>(kind: S, error: BoxError) -> Self {
-        let kind = kind.to_string();
-
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{kind}: {error:#?}"))
-    }
-
-    /// Return a rejected response containing just the detailed error information.
-    pub fn error(error: BoxError) -> Self {
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{error:#?}"))
-    }
-}
-
-impl<E: Error + Send + Sync + 'static> From<E> for ProposalResponse {
-    fn from(error: E) -> Self {
-        Self::error(error.into())
+    ///
+    /// Note: Error kind is currently ignored to match zcashd error format (`kebab-case` string).
+    pub fn rejected<S: ToString>(_kind: S, error: BoxError) -> Self {
+        // Make error `kebab-case` to match zcashd format.
+        let error_kebab1 = format!("{error:?}")
+            .replace(|c: char| !c.is_alphanumeric(), "-")
+            .to_ascii_lowercase();
+
+        // Remove consecutive `-` characters if any.
+        let regex1 = regex::Regex::new(r"-+").expect("valid regex");
+        let error_kebab2 = regex1.replace_all(&error_kebab1, "-");
```

Comment:
I think adding the `regex` crate is okay, but if we want to remove it, we could use [`as_mut_vec`](https://doc.rust-lang.org/std/string/struct.String.html#method.as_mut_vec), [`dedup`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup), and [`trim_matches`](https://doc.rust-lang.org/std/primitive.str.html#method.trim_matches) instead. 

`as_mut_vec` is unsafe because it might add invalid characters but in this case we would only be removing characters.

Reply:
We avoid using unsafe code when there is a safe alternative.

Here is a safe alternative to `as_mut_vec`:
```rust
let mut error: Vec<char> = error.chars().collect();
error.dedup();
let error: String = error.into_iter().collect();
```

Reply:
The only problem here and the reason we still have a regex left is with the `dedup()`. The method remove all the repeated characters so an error will end up looking as: 

`io-eror-kind-unexpectedeof-mesage-failed-to-fil-whole-bufer`

So i didn't applied it.

Reply:
That worked, thanks! https://github.com/ZcashFoundation/zebra/pull/6044/commits/a8b4d718bb8e5f1950c620de783c0f3f9d11d8d7